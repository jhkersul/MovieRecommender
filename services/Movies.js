import moviesData from '../data/movies.json';
import linksData from '../data/links.json';

/**
 * Returns the movies data
 */
export const getMovies = () => moviesData;

/**
 * Returns the links data
 */
export const getLinks = () => linksData;

/**
 * Returns the The Movie Database ID from a movie
 * @param {Number} movieID The current movieID to get the TmdbId
 */
export const getMovieTmdbId = (movieID) => {
  // Filtering by movieId
  const filteredItems = getLinks().filter(link => link.movieId === movieID);
  // Checking if the filter found any item
  if (filteredItems.length > 0) {
    // Returning the tmdbId
    const tmdbId = filteredItems[0].tmdbId;
    return tmdbId;
  }
  return null;
};
