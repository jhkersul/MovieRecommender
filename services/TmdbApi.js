import config from '../config';
import { request, GET } from './Network';

/**
 * The TMDB API Key from config file
 */
export const TMDB_API_KEY = config.TMDB_API_V3_KEY;

/**
 * The URL from TMDB APi
 */
export const API_TMDB_URL = config.API_TMDB_URL;

export const getMovieById = tmdbId => (
  new Promise((resolve, reject) => {
    if (typeof tmdbId === 'undefined' || tmdbId === null) reject('Invalid tmdbId.');

    const params = {
      api_key: TMDB_API_KEY,
    };

    request(GET, `${API_TMDB_URL}/movie/${tmdbId}`, params)
      .then((response) => {
        // If the response has a status_code, it means it's an error
        if (response.status_code) {
          reject(response);
          return;
        }
        resolve(response);
      })
      .catch(error => reject(error));
  })
);
