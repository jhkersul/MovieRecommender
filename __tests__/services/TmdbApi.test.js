import { getMovieById } from '../../services/TmdbApi';


test('Movie info must be snapshot', () => {
  expect.assertions(1);
  return getMovieById(8844).then((response) => {
    expect(response).toMatchSnapshot();
  });
});

