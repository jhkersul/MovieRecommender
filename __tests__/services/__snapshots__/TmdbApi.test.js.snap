// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie info must be snapshot 1`] = `
Object {
  "adult": false,
  "backdrop_path": "/52lVqTDhIeNTjT7EiJuovXgw6iE.jpg",
  "belongs_to_collection": null,
  "budget": 65000000,
  "genres": Array [
    Object {
      "id": 12,
      "name": "Adventure",
    },
    Object {
      "id": 14,
      "name": "Fantasy",
    },
    Object {
      "id": 10751,
      "name": "Family",
    },
  ],
  "homepage": "",
  "id": 8844,
  "imdb_id": "tt0113497",
  "original_language": "en",
  "original_title": "Jumanji",
  "overview": "When siblings Judy and Peter discover an enchanted board game that opens the door to a magical world, they unwittingly invite Alan -- an adult who's been trapped inside the game for 26 years -- into their living room. Alan's only hope for freedom is to finish the game, which proves risky as all three find themselves running from giant rhinoceroses, evil monkeys and other terrifying creatures.",
  "popularity": 17.428288,
  "poster_path": "/vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg",
  "production_companies": Array [
    Object {
      "id": 559,
      "name": "TriStar Pictures",
    },
    Object {
      "id": 2550,
      "name": "Teitler Film",
    },
    Object {
      "id": 10201,
      "name": "Interscope Communications",
    },
  ],
  "production_countries": Array [
    Object {
      "iso_3166_1": "US",
      "name": "United States of America",
    },
  ],
  "release_date": "1995-12-15",
  "revenue": 262797249,
  "runtime": 104,
  "spoken_languages": Array [
    Object {
      "iso_639_1": "en",
      "name": "English",
    },
    Object {
      "iso_639_1": "fr",
      "name": "Fran√ßais",
    },
  ],
  "status": "Released",
  "tagline": "Roll the dice and unleash the excitement!",
  "title": "Jumanji",
  "video": false,
  "vote_average": 6.9,
  "vote_count": 2262,
}
`;
